src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+using Raven.Client.Documents.Conventions;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Commands.Batches
+{
+    public class PutAttachmentCommandData : ICommandData
+    {
+        public PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, long? etag)
+        {
+            if (string.IsNullOrWhiteSpace(documentId))
+                throw new ArgumentNullException(nameof(documentId));
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            Key = documentId;
+            Name = name;
+            Stream = stream;
```

Comment:
I'm not sure if we should force the stream position to be `0`. Maybe just keep the original position we got here and use it once we need to resend the stream - that's how ravenfs client behaves currently.